src/views/HomeView.vue

```diff
@@ -37,19 +31,28 @@ export default {
       return this.$store.getters['search/getItems'];
     }
   },
-  mounted() {
-    this.fetchMovies();
+  beforeRouteEnter(to, from, next) {
+      next(vm => {
+        if (vm.$route.path !== '/') {
+          vm.fetchMovies(vm.$route.path)
+        }
+    })
   },
   methods: {
-    fetchMovies() {
-      switch(this.selectValue) {
-        case 'top-rated':
+    fetchMovies(path) {
+      switch(path) {
+        case '/top-rated':
           this.$store.dispatch('search/fetchTopRatedMovies');
           break;
-        case 'upcoming':
+        case '/upcoming':
           this.$store.dispatch('search/fetchUpcomingMovies');
           break;
-        case 'popular':
+        case '/popular':
+          this.$store.dispatch('search/fetchPopularMovies');
+          break;
+        case '/search':
+          break;
+        case '/':
```

Comment:
This case is not needed. Because this method is not triggered when `path == '/'`.
Also `/search` case can be removed. Then update `default` case to make a `console.info('Un-handled path: ', path)`